creator-node/src/routes/files.js

```diff
@@ -247,46 +254,97 @@ module.exports = function (app) {
     // Do not act as a public gateway. Only serve IPFS files that are tracked by this creator node.
     const dirCID = req.params.dirCID
     const filename = req.params.filename
+    const ipfsPath = `${dirCID}/${filename}`
 
     const queryResults = await models.File.findOne({ where: {
-      multihash: dirCID,
-      type: 'dir'
+      dirMultihash: dirCID,
+      // All other file types are valid and can be served through this route.
+      type: { [models.Sequelize.Op.ne]: 'dir' }, // Op.ne = notequal
+      fileName: filename
     } })
     if (!queryResults) {
-      return sendResponse(req, res, errorResponseNotFound(`No dir entry found for provided dirCID: ${dirCID}`))
+      return sendResponse(
+        req,
+        res,
+        errorResponseNotFound(`No valid file found for provided dirCID: ${dirCID} and filename: ${filename}`)
+      )
     }
+    const parentStoragePath = queryResults.storagePath.split('/').slice(0, -1).join('/')
```

Comment:
can you add a comment as to what this is doing. also can we stick to calling it dirStoragePath?